common/src/types/v0/transport/cluster_agent.rs

```diff
@@ -26,3 +26,40 @@ impl NodeAgentInfo {
         self.endpoint
     }
 }
+
+/// Failed NVMe path.
+#[derive(Debug, Clone)]
+pub struct FailedPath {
+    target_nqn: String,
+}
+
+impl FailedPath {
+    pub fn new(target_nqn: String) -> Self {
+        Self { target_nqn }
+    }
+
+    pub fn target_nqn(&self) -> &str {
+        &self.target_nqn
+    }
+}
+
+/// Report failed NVMe paths.
+#[derive(Debug)]
+pub struct ReportFailedPaths {
+    node: String,
+    failed_paths: Vec<FailedPath>,
+}
+
+impl ReportFailedPaths {
+    pub fn new(node: String, failed_paths: Vec<FailedPath>) -> Self {
+        Self { node, failed_paths }
+    }
+
+    pub fn node_name(&self) -> &str {
+        &self.node
+    }
+
+    pub fn failed_paths(&self) -> Vec<FailedPath> {
+        self.failed_paths.clone()
```

Comment:
Why cloning Vec here? You could return &[FailedPath].

Reply:
Yup, indeed.